### PR TITLE
Support manual application of StringStyle transforms via attributes dictionary

### PR DESCRIPTION
**Pull request by jvisenti**
Addresses #297

### CODE REVIEW

**Comment by ZevEisenberg** [comment]
@jvisenti have you been using this in production code? Think it's up to snuff for merging?

**Comment by ZevEisenberg** [comment]
Sounds reasonable. We should rebase this at some point, or port the changes over, now that #296 has landed. Rebase looks a little hairy, so a by-hand port might make more sense.

File: Sources/Transform.swift [hunk 1]
```diff
@@ -25,7 +25,7 @@ public enum Transform {
     case capitalizedWithLocale(Locale)
     case custom(TransformFunction)
 
-    var transformer: TransformFunction {
+    public var transformer: TransformFunction {
```
Comment (cjamie) [inline Sources/Transform.swift]: this change shouldn't be necessary

File: Sources/Transform.swift [hunk 2]
```diff
@@ -40,3 +40,76 @@ public enum Transform {
...
+            case lowercase
+            case uppercase
+            case capitalized
+            case custom
+
+        }
+
+        internal var name: Name {
+            switch self {
+            case .lowercase, .lowercaseWithLocale: return Name.lowercase
```
Comment (cjamie) [inline Sources/Transform.swift]: would like to take advantage of swift's type inference: 
```suggestion 
            case .lowercase, .lowercaseWithLocale: return .lowercase
```

File: Tests/AttributedStringStyleTests.swift [hunk 3]
```diff
@@ -37,7 +37,7 @@ class StringStyleTests: XCTestCase {
     #if os(iOS) || os(tvOS)
     func testTextStyle() {
         let style = StringStyle(.textStyle(titleTextStyle))
-        for (style, fullStyle) in additiviePermutations(for: style) {
+        for (style, fullStyle) in additivePermutations(for: style) {
```
Comment (cjamie) [inline Tests/AttributedStringStyleTests.swift]: while we're here, why don't we prefer forEach instead? 
```suggestion 
       additivePermutations(for: style).forEach { (style, fullStyle) in
```

File: Sources/UIKit/EmbeddedTransformation.swift [hunk 4]
```diff
@@ -60,7 +60,7 @@ internal enum EmbeddedTransformationHelpers {
         let representations = styleAttributes[BonMotTransformationsAttributeName] as? [StyleAttributes] ?? []
         let results: [T?] = representations.map { representation in
             for type in embeddedTransformationTypes {
-                if let transformation = type.from(dictionary: representation) as? T {
+                if let transformation = type.init(dictionary: representation) as? T {
```
Comment (cjamie) [inline Sources/UIKit/EmbeddedTransformation.swift]: for line 61, we can change this to a compactMap so we don't need to go through the collection again, and with flattop
```suggestion
         let results: [T] = representations.compactMap { representation in
```

File: Tests/TransformTests.swift [hunk 5]
```diff
@@ -183,4 +183,25 @@ class TransformTests: XCTestCase {
...
+            .adapt(.control)
+        )
+
+        let attributes = style.attributes
+        XCTAssertNotNil(attributes[BonMotTransformationsAttributeName])
+
+        let transform = (attributes[BonMotTransformationsAttributeName] as? StyleAttributes).flatMap(Transform.init)
+        XCTAssertNotNil(transform)
+
+        if let transformer = transform?.transformer {
```
Comment (cjamie) [inline Tests/TransformTests.swift]: in tests, I think its ok to force unwrap, so we can just get a failing test

**Comment by chrisballinger** [comment]
Looks like this has been hanging open for a few years. Going to close this for now, but if someone needs this feature and wants to resurrect it, go ahead!

**Pull request closed without merging**